source/common/config/grpc_mux_impl.cc

```diff
@@ -149,16 +145,28 @@ void GrpcMuxImpl::onDiscoveryResponse(
     return;
   }
 
-  if (apiStateFor(type_url).watches_.empty()) {
+  ApiState& api_state = apiStateFor(type_url);
+
+  if (message->has_control_plane()) {
+    control_plane_stats.identifier_.set(message->control_plane().identifier());
+  }
+
+  if (message->control_plane().identifier() != api_state.control_plane_identifier_) {
+    api_state.control_plane_identifier_ = message->control_plane().identifier();
```

Comment:
Why is this tracked on a per type URL basis? Shouldn't any reconnect cause the ID to change for all subsequent messages? Or do you think that the control plane identifier might in some cases change on a per type URL basis?

Reply:
The latter. I can imagine a pass-through frontend that serves ADS by aggregating xDS streams from multiple backends, where the backend identifier is more useful than the frontend. Also Envoy itself can connect to different servers for LDS, CDS, EDS, etc, right?

Reply:
Not in a single gRPC mux; the gRPC mux is governing a single xDS connection (of which there may be > 1).

I guess this is OK, it's not really something I think is a likely situation but since it's technically possible by virtue of the protos we can go this way..